src/dotnet/commands/dotnet-compile/ManagedCompiler.cs

```diff
@@ -181,31 +184,15 @@ public override bool Compile(ProjectContext context, CompilerCommandApp args)
 
             _scriptRunner.RunScripts(context, ScriptNames.PreCompile, contextVariables);
 
-            var result = _commandFactory.Create($"compile-{compilerName}", new[] { "@" + $"{rsp}" })
-                .OnErrorLine(line =>
-                {
-                    var diagnostic = ParseDiagnostic(context.ProjectDirectory, line);
-                    if (diagnostic != null)
-                    {
-                        diagnostics.Add(diagnostic);
-                    }
-                    else
-                    {
-                        Reporter.Error.WriteLine(line);
-                    }
-                })
-                .OnOutputLine(line =>
-                {
-                    var diagnostic = ParseDiagnostic(context.ProjectDirectory, line);
-                    if (diagnostic != null)
-                    {
-                        diagnostics.Add(diagnostic);
-                    }
-                    else
-                    {
-                        Reporter.Output.WriteLine(line);
-                    }
-                }).Execute();
+            // Cache the reporters before invoking the command in case it is a built-in command, which replaces
```

Comment:
Can this be handled in the command factory? Feels like it would be cleaner than having to do this wherever we use the factory.